common/thorhelper/thorcommon.cpp

```diff
@@ -1227,28 +1215,26 @@ class CRowStreamReader : public CSimpleInterfaceOf<IExtRowStream>
     {
         return progress;
     }
-    virtual bool setFiltersAndTranslation(IConstArrayOf<IFieldFilter> &filters, IOutputMetaData *_actualFormat, IOutputMetaData *projectedFormat, RecordTranslationMode translationMode, unsigned expectedCrc, unsigned daliCrc) override
+    virtual void setFilters(IConstArrayOf<IFieldFilter> &filters)
     {
-        actualFormat.set(_actualFormat);
-        bool translate = getTranslators(translator, keyedTranslator, "rowstream", expectedFormat, actualFormat, projectedFormat, translationMode, expectedCrc, daliCrc);
-        if (!actualFormat)
-            actualFormat.set(expectedFormat);
-        actual = &actualFormat->queryRecordAccessor(true);
-        actualFilter.clear();
+        if (filterRow)
+        {
+            delete filterRow;
+            filterRow = nullptr;
+            actualFilter.clear();
+        }
         if (filters.ordinality())
         {
             actualFilter.appendFilters(filters);
-            if (keyedTranslator)
-                keyedTranslator->translate(actualFilter);
+            if (translatorContainer)
+            {
+                const IKeyTranslator *keyedTranslator = translatorContainer->queryKeyedTranslator();
+                if (keyedTranslator)
+                    keyedTranslator->translate(actualFilter);
```

Comment:
If this is called twice, will the filters get translated twice?  Should it be cloning the filters instead?

Reply:
The filters are parsed in and effectively cloned (actualFilter is cleared and the passed in filters are added to actualFilter. So recalling setFilters() will be translating a new actualFilter.

Reply:
I misremembered, it doesn't modify the existing IFieldfilter, so not a problem.